src/software/amazon/ion/impl/IonReaderTextSystemX.java

```diff
@@ -72,6 +73,46 @@ public IonSystem getSystem()
         return _system;
     }
 
+    // TODO getIntegerType() is duplicated in IonReaderBinarySystemX. It could
+    // be consolidated into a single location, but that would have to be part
```

Comment:
You could at least consolidate the `AS_TYPE` → `IntegerType` logic into a static method on `AS_TYPE`.